preload_associations.go

```diff
@@ -471,6 +471,7 @@ func preloadManyToMany(tx *Connection, asoc *AssociationMetaInfo, mmi *ModelMeta
 	if err != nil {
 		return err
 	}
+	defer rows.Close()
```

Comment:
also redundant.

Reply:
This is important if we return an error early.

Reply:
If we return early like in line 517, we leak the rows. I don't think rolling back the transaction necessarily fixes that. Either way, it's an incorrect usage of the `database/sql` API.

I fixed this function by searching for `rows.Next()`.

Reply:
First, as I already told you, I don't mind this part, it could be fine.
However, even if the loop returns before closing the rows, the rows is within a transaction and the transaction is soon rolled back, so a connection leak never happens because the database server and the client do not need to wait for rows to be used since the whole transaction was already canceled. This is a correct usage of the `database/sql`.

(The reason I asked whether you use it or not is that this feature is not solid yet and I found a critical issue in the code during the review for your PR. It could be fine if you don't use it.)